pecos/core/xmc/linear_solver.hpp

```diff
@@ -133,9 +412,23 @@ struct SVMWorker {
             solve_l2r_l1l2_svc(X, seed);
         } else if(param.solver_type == L2R_LR_DUAL) {
             solve_l2r_lr(X, seed);
+        } else if (param.solver_type == L2R_L2LOSS_SVC_PRIMAL) {
+            solve_l2r_l2_svc_primal(X, seed);
         }
     }
 
+    template <typename MAT>
+    void solve_l2r_l2_svc_primal(const MAT &X, int seed) {
+        dvec_t tmp_y, tmp_c;
+        tmp_y.resize(y_size, 0);
+        tmp_c.resize(y_size, 0);
+        
+        l2r_l2_svc_fun<MAT, value_type, SVMWorker> fun_obj(&param, X, this);
+        NEWTON<MAT, value_type, SVMWorker> newton_obj(&fun_obj);
+        dvec_wrapper_t curr_w(w);
```

Comment:
The inconsistancy is from here: you need to set initial value (default 0) for `curr_w` and `b`.